src/nvim/ex_cmds.c

```diff
@@ -4822,8 +4822,8 @@ static void helptags_one(char_u *dir, char_u *ext, char_u *tagfname,
 
   // Find all *.txt files.
   size_t dirlen = STRLEN(dir);
-  STRCPY(NameBuff, dir);
-  STRCAT(NameBuff, "/**/*");
+  STRLCPY(NameBuff, dir, sizeof(NameBuff));
+  STRCAT(NameBuff, "/**/*");  // NOLINT
```

Comment:
regex testers (for instance at regex101.com) show that for strings as small as 30chars, this can take about `6.5 * 10^4` steps. The regex looks harmless and it is possible to test for the equivalent efficiently in a single pass, but looks like there's a lot of backtracking that's happening.